csharp/src/Google.Protobuf/Reflection/MessageDescriptor.cs

```diff
@@ -62,6 +62,7 @@ public sealed class MessageDescriptor : DescriptorBase
         private readonly IList<EnumDescriptor> enumTypes;
         private readonly IList<FieldDescriptor> fieldsInDeclarationOrder;
         private readonly IList<FieldDescriptor> fieldsInNumberOrder;
+        private readonly IDictionary<string, FieldDescriptor> jsonFieldMap;
```

Comment:
should we use Lazy (https://msdn.microsoft.com/en-us/library/dd642331(v=vs.110).aspx)?
It's not clear that everyone will use JsonParsing and for mobile (Xamarin), lazy initialization could decrease app startup time. (AFAIK, on Android this has been a concern).